testkit-java/src/main/java/com/akkaserverless/javasdk/testkit/ActionResult.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.testkit;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import com.akkaserverless.javasdk.ServiceCall;
+import java.util.concurrent.CompletableFuture;
+import com.akkaserverless.javasdk.action.Action;
+
+public interface ActionResult<T> {
+
+  /** @return true if the call had an effect with a reply, false if not */
+  boolean isReply();
+
+  T getReplyMessage();
+
+  /** @return true if the call was forwarded, false if not */
+  boolean isForward();
+
+  ServiceCall getForwardServiceCall();
+
+  // TODO rewrite name. Async doesn't seem very descriptive. Are the rest sync?
+  /** @return true if the call was async, false if not */
+  boolean isAsync();
+
+  CompletableFuture<Action.Effect<T>> getAsyncEffect();
```

Comment:
This one is tricky, as we are back to no user API for this one, I wonder if it perhaps could be `CompletionStage<ActionResult<T>>` , that way a user can block/wait for it in the test and _then_ inspect that result.

Another option could perhaps be to block in here for async effects to complete, in the normal calls if it is async, but then there is the question about what timeout to use, so maybe better to hand the waiting off to the user/test logic.

Note that the return type should be `CompletionStage`, the `CompletableFuture` type is the one a user can complete (like `Promise`).

Reply:
done